fs-index/src/index.rs

```diff
@@ -474,4 +474,109 @@ impl<Id: ResourceId> ResourceIndex<Id> {
+        /*
+           # Given:
+            - The file system contains resources with duplicate IDs
+              (e.g., hash collisions or files with the same content)
+            - The index is up-to-date with the file system except for the new
+              resource
+            - The resource is not already in the index
+
+           # Problem:
+           - We only know the ID of the resource
+           - We need to get the path of the resource so to add it to the index
+           - To get the path, we need to scan the file system for the resource
+             that fulfills theses conditions:
+                - The resource has the given ID
+                - The resource path is not in the index
+            To do this, we need to iterate over all the resources in the file
+            system and check if the resource has the given ID and is not in the
+            index (complexity: O(n)).
+           - Track API should be O(1) or O(log n) if possible because it is
+             used for fine-grained control over tracking changes in the index
+        */
+
+        unimplemented!("Track addition is not implemented yet");
+    }
+
+    /// Track the removal of a resource
+    ///
+    /// This method accepts a resource ID and returns `Result::Ok` if the
+    /// resource was successfully removed from the index.
+    ///
+    /// **Note**: The called must ensure that:
+    /// - The index is up-to-date with the file system except for the removed
+    ///  resource
+    /// - The resource is already in the index
+    pub fn track_removal(&mut self, _id: Id) -> Result<()> {
+        /*
+           # Given:
+            - The file system contains resources with duplicate IDs
+            - The index is up-to-date with the file system except for the
+              removed resource
+            - The resource is no more in the file system
+
+           # Problem:
+           - We only know the ID of the resource
+           - We need to get the path of the resource so to remove it from the
+             index maps.
+           - Since the resource is already in the index, we can call `get_resources_by_id`
+             to get the possible paths of the resource and then find the path that is
+             not in the file system (worst case complexity: O(n) but this is
+             super exagerated. In practice, it should be so much faster since
+             not all the resources are duplicates).
+        */
+
+        unimplemented!("Track removal is not implemented yet");
+    }
+
+    /// Track the modification of a resource
+    ///
+    /// This method accepts a resource ID and returns `Result::Ok` if the
+    /// resource was successfully updated in the index.
+    ///
+    /// **Note**: The called must ensure that:
+    /// - The index is up-to-date with the file system except for the updated
+    ///  resource
+    /// - The resource is already in the index
+    pub fn track_modification(&mut self, _id: Id) -> Result<()> {
```

Comment:
Original motivation was here:
- https://github.com/ARK-Builders/arklib/issues/38

The use-case was updating the resource in one app, after its modification in another app.

But now I think, this function is not needed, because:
1. The editor app can have different flow, like opening a resource by one path, and saving the modified version to another path. Reactive API is much more robust here, especially with `notify`. We should ensure that `update_all` is efficient, too. We had a bug in there causing very slow re-indexing every time a single update was made. That's why the `track_update` appeared originally.
2. If the editor overwrites the resource by same path, then it still can use combination of `track_deletion` and `track_addition`.

Reply:
**tl;dr**: consider removing